File: x/psets/2/readability/readability.c
/*
    JONASH MARCELINO
    CS50
*/
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

// Function declaration
int count_letters(string);
int count_words(string);
int count_sentences(string);

int main(void)
{
    // Get text from user
    string text = get_string("Text: ");

    // Counters
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calcualte the index
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print Grade
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Function that counts letter
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

// Function that counts word
int count_words(string text)
{
    int words = 1;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

// Function that counts sentence
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: x/psets/6/sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    """ Main Function """
    # Get input from user
    text = get_string("Text: ")

    # Counters
    letters = count_letters(text)
    words = len(text.split())
    sentences = count_sentences(text)

    # Calculate the index
    ltr = letters / words * 100
    snt = sentences / words * 100
    index = int(round(0.0588 * ltr - 0.296 * snt - 15.8))

    # Print Grade
    if index < 1:
        print("Before Grade 1")
    elif index >= 17:
        print("Grade 16+")
    else:
        print(f"Grade", index)


def count_letters(txt):
    """ Function that counts letter in text """
    letter_ctr = 0
    for c in txt:
        if c.isalpha():
            letter_ctr += 1
    return letter_ctr


def count_sentences(txt):
    """ Function that counts sentences in text """
    sentence_ctr = 0
    for c in txt:
        if c in ['.', '!', '?']:
            sentence_ctr += 1
    return sentence_ctr


if __name__ == '__main__':
    main()